From 52a1a2ce6633b8cee8968b8ef6f95996ce4bedc5 Mon Sep 17 00:00:00 2001
From: mkundaragi <mkundaragi@75f.io>
Date: Wed, 6 Jul 2022 06:07:09 +0000
Subject: [PATCH 11/15] Merged PR 4320: SN CPU profile fan low fix

Turning on Fan 1 when cooling stage 2 is turned on

Related work items: #10826
---
 .../io/logic/bo/building/sscpu/ConventionalUnitProfile.java   | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/logic/src/main/java/a75f/io/logic/bo/building/sscpu/ConventionalUnitProfile.java b/logic/src/main/java/a75f/io/logic/bo/building/sscpu/ConventionalUnitProfile.java
index 46e4b7669..b476822bc 100644
--- a/logic/src/main/java/a75f/io/logic/bo/building/sscpu/ConventionalUnitProfile.java
+++ b/logic/src/main/java/a75f/io/logic/bo/building/sscpu/ConventionalUnitProfile.java
@@ -245,6 +245,10 @@ public class ConventionalUnitProfile extends ZoneProfile {
                         if(getCmdSignal("cooling and stage2",node) == 0)
                             setCmdSignal("cooling and stage2", 1.0, node);
                         relayStages.put("CoolingStage2",1);
+                        if(isFanStage1Enabled && getCmdSignal("fan and stage1", node) == 0) {
+                                setCmdSignal("fan and stage1", 1, node);
+                                relayStages.put("FanStage1",1);
+                        }
                         if(((isFanStage2Enabled  && ((fanSpeed == AUTO) || isFanSpeedHigh) ) ||  (occupied && enableFanStage1DuringOccupied && isFanSpeedHigh))&& (fanHighType == SmartStatFanRelayType.FAN_STAGE2)){
                             relayStages.put("FanStage2",1);
                             if(getCmdSignal("fan and stage2",node) == 0)
-- 
2.31.0.windows.1

